gorp_test.go

```diff
@@ -404,6 +404,282 @@ type PersistentUser struct {
 	PassedTraining bool
 }
 
+type TenantDynamic struct {
+	Id       int64 `db:"id"`
+	Name     string
+	Address  string
+	curTable string `db:"-"`
+}
+
+func (curObj *TenantDynamic) TableName() string {
+	return curObj.curTable
+}
+func (curObj *TenantDynamic) SetTableName(tblName string) {
+	curObj.curTable = tblName
+}
+
+var dynTableInst1 = TenantDynamic{curTable: "t_1_tenant_dynamic"}
+var dynTableInst2 = TenantDynamic{curTable: "t_2_tenant_dynamic"}
+
+func dynamicTablesTest(t *testing.T, dbmap *gorp.DbMap) {
+
+	dynamicTablesTestTableMap(t, dbmap, &dynTableInst1)
+	dynamicTablesTestTableMap(t, dbmap, &dynTableInst2)
+
+	// TEST - dbmap.Insert using dynTableInst1
+	dynTableInst1.Name = "Test Name 1"
+	dynTableInst1.Address = "Test Address 1"
+	err := dbmap.Insert(&dynTableInst1)
+	if err != nil {
+		t.Errorf("Errow while saving dynTableInst1. Details: %v", err)
+	}
+
+	// TEST - dbmap.Insert using dynTableInst2
+	dynTableInst2.Name = "Test Name 2"
+	dynTableInst2.Address = "Test Address 2"
+	err = dbmap.Insert(&dynTableInst2)
+	if err != nil {
+		t.Errorf("Errow while saving dynTableInst2. Details: %v", err)
+	}
+
+	dynamicTablesTestSelect(t, dbmap, &dynTableInst1)
+	dynamicTablesTestSelect(t, dbmap, &dynTableInst2)
+	dynamicTablesTestSelectOne(t, dbmap, &dynTableInst1)
+	dynamicTablesTestSelectOne(t, dbmap, &dynTableInst2)
+	dynamicTablesTestGetUpdateGet(t, dbmap, &dynTableInst1)
+	dynamicTablesTestGetUpdateGet(t, dbmap, &dynTableInst2)
+	dynamicTablesTestDelete(t, dbmap, &dynTableInst1)
+	dynamicTablesTestDelete(t, dbmap, &dynTableInst2)
+
+}
+
+func dynamicTablesTestTableMap(t *testing.T,
+	dbmap *gorp.DbMap,
+	inpInst *TenantDynamic) {
+
+	tableName := inpInst.TableName()
+
+	tblMap, err := dbmap.TableForDynamic(tableName, true)
+	if nil != err {
+		t.Errorf("Error while searching for tablemap for tableName: %v, Error:%v", tableName, err)
+	}
+	if nil == tblMap {
+		t.Errorf("Unable to find tablemap for tableName:%v", tableName)
+	}
+}
+
+func dynamicTablesTestSelect(t *testing.T,
+	dbmap *gorp.DbMap,
+	inpInst *TenantDynamic) {
+
+	// TEST - dbmap.Select using inpInst
+
+	// read the data back from dynInst to see if the
+	// table mapping is correct
+	var dbTenantInst1 = TenantDynamic{curTable: inpInst.curTable}
+	selectSQL1 := "select * from " + inpInst.curTable
+	dbObjs, err := dbmap.Select(&dbTenantInst1, selectSQL1)
+	if err != nil {
+		t.Errorf("Errow in dbmap.Select. SQL: %v, Details: %v", selectSQL1, err)
+	}
+	if nil == dbObjs {
+		t.Errorf("Nil return from dbmap.Select")
+	} else {
```

Comment:
I'm also updating this - `t.Fatalf` will perform the same comparison but halt the test immediately, making this `else` statement unnecessary.